include/crypto/riscv_arch.h

```diff
@@ -36,26 +45,50 @@ struct RISCV_capability_s {
     const char *name;
     size_t index;
     size_t bit_offset;
+# ifdef OSSL_RISCV_HWPROBE
+    int32_t hwprobe_key;
+    uint64_t hwprobe_value;
+# endif
 };
 
-# define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX) +1
+# define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX, \
+                          OSSL_RISCV_HWPROBE_KEY, OSSL_RISCV_HWPROBE_VALUE) +1
 extern const struct RISCV_capability_s RISCV_capabilities[
 # include "riscv_arch.def"
 ];
 
 # ifdef OPENSSL_RISCVCAP_IMPL
-#  define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX) \
+#  ifdef OSSL_RISCV_HWPROBE
+#  define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX,     \
+                           HWPROBE_KEY, HWPROBE_VALUE) \
+    { #NAME, INDEX, BIT_INDEX, HWPROBE_KEY, HWPROBE_VALUE },
+#  else
+#  define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX,     \
+                           HWPROBE_KEY, HWPROBE_VALUE) \
     { #NAME, INDEX, BIT_INDEX },
+#  endif
 const struct RISCV_capability_s RISCV_capabilities[] = {
 #  include "riscv_arch.def"
 };
 # endif
 
-# define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX) +1
+# define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX, \
+                          HWPROBE_KEY, HWPROBE_VALUE) +1
 static const size_t kRISCVNumCaps =
 # include "riscv_arch.def"
 ;
 
+# ifdef OSSL_RISCV_HWPROBE
+/*
+ * Content is an array of { hwprobe_key, 0 } where
+ * hwprobe_key is copied from asm/hwprobe.h.
+ * It should be updated along with riscv_arch.def.
+ */
+#  define OSSL_RISCV_HWPROBE_PAIR_COUNT 1
+#  define OSSL_RISCV_HWPROBE_PAIR_CONTENT \
+   { 4, 0 },
```

Comment:
It's better to replace `4` here with `RISCV_HWPROBE_KEY_IMA_EXT_0` with `#define RISCV_HWPROBE_KEY_IMA_EXT_0	4` somewhere.